src/main/java/de/azapps/kafkabackup/common/topic/TopicConfigurationService.java

```diff
@@ -27,51 +27,51 @@
   Semaphore configurationCheckSemaphore = new Semaphore(1);
 
   public void runTopicConfigurationCheck() {
```

Comment:
Let's move this asynchronous method to BackupSinkTask class. This way service will be synchronous and in sink task we decide to make it asynchronous.
WDYT?

Reply:
Very nice! Will do.